recipes/gsl/all/conanfile.py

```diff
@@ -0,0 +1,158 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from contextlib import contextmanager
+import os
+
+required_conan_version = ">=1.33.0"
+
+class GslConan(ConanFile):
+    name = "gsl"
+    description = "GNU Scientific Library"
+    homepage = "https://www.gnu.org/software/gsl"
+    topics = ("numerical", "math", "random", "scientific")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "GPL-3.0-or-later"
+
+    settings = "os", "compiler", "build_type", "arch"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True
+    }
+
+    exports_sources = "patches/*"
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
+                  destination=self._source_subfolder)
+
+    def build_requirements(self):
```

Comment:
```suggestion
    def build_requirements(self):
        if self.settings.os == "Macos" and self.settings.arch == "armv8":
            # Attempting to use @rpath without CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG being
            # set. This could be because you are using a Mac OS X version less than 10.5
            # or because CMake's platform configuration is corrupt.
            self.build_requires("cmake/3.20.1")
```